helpers/copy_tree.sh

```diff
@@ -27,6 +29,20 @@ get_closest() {
     fi
 }
 
+BUILD_HOST=
+BUILD_USER=
+fixup_buildprop() {
+    if [ -f "$1" ]; then
+        # pray that these are unique enough.
+        if [ ! -z "$BUILD_HOST" ] && [ ! -z "$BUILD_HOST_REPLACEMENT" ]; then
+            sed -i "s/$BUILD_HOST/$BUILD_HOST_REPLACEMENT/g" "$1"
+        fi
+        if [ ! -z "$BUILD_USER" ] && [ ! -z "$BUILD_USER_REPLACEMENT" ]; then
```

Comment:
shellcheck prefers this to be `-n` :)
```suggestion
        if [ -n "$BUILD_USER" ] && [ -n "$BUILD_USER_REPLACEMENT" ]; then
```